Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py

```diff
@@ -169,29 +175,50 @@ def get_outbound_ioc_values(on_demand, limit, offset, indicator_query='', out_fo
     last_offset = last_update_data.get('last_offset')
     last_format = last_update_data.get('last_format')
     last_query = last_update_data.get('last_query')
+    current_iocs = last_update_data.get('current_iocs')
+
     # on_demand ignores cache
     if on_demand:
-        values_str = get_ioc_values_str_from_context()
+        if out_format != last_format or limit != last_limit or offset != last_offset:
+            values_str = get_ioc_values_str_from_context(current_iocs, out_format, limit, offset)
+
+        else:
+            values_str = get_ioc_values_str_from_context()
+
     else:
         if last_update:
             # takes the cache_refresh_rate amount of time back since run time.
             cache_time, _ = parse_date_range(cache_refresh_rate, to_timestamp=True)
-            if last_update <= cache_time or last_limit != limit or last_offset != offset or \
-                    last_format != out_format or indicator_query != last_query:
+            if last_update <= cache_time or last_limit != limit or \
+                    last_offset != offset or last_format != out_format or indicator_query != last_query:
                 values_str = refresh_outbound_context(indicator_query, out_format, limit=limit, offset=offset)
             else:
                 values_str = get_ioc_values_str_from_context()
         else:
             values_str = refresh_outbound_context(indicator_query, out_format, limit=limit, offset=offset)
+
     return values_str
 
 
-def get_ioc_values_str_from_context() -> str:
+def get_ioc_values_str_from_context(iocs=None, new_format: str = FORMAT_TEXT,
+                                    limit: int = 10000, offset: int = 0) -> str:
     """
     Extracts output values from cache
     """
-    cache_dict = demisto.getIntegrationContext()
-    return cache_dict.get(CTX_VALUES_KEY, '')
+    if iocs:
+        if offset > len(iocs):
+            return ''
+
+        iocs = iocs[offset: limit + offset]
+        returned_dict = create_values_out_dict(iocs, new_format)
+        current_cache = demisto.getIntegrationContext()
+        current_cache['last_output'] = returned_dict
+        demisto.setIntegrationContext(current_cache)
+
+    else:
+        returned_dict = demisto.getIntegrationContext().get('last_output')
```

Comment:
can we have a case in which in the first fetch this will be empty?

Reply:
yes it can happen if the query returns nothing or if the offset requested is larger than the amount brought back.

Reply:
Then it could break the return line(line 221), you should have a default value of {}